timex/timex.go

```diff
@@ -0,0 +1,67 @@
+// Package timex provides stdlib time package related utilities.
+//
+// It exports type Ticker which works like stdlib time.Ticker but closes
+// its channel on stop.
+package timex
+
+import (
+	"time"
+)
+
+// A Ticker holds a channel that delivers 'ticks' of a clock at
+// intervals, similar to time.Ticker from the standard library.
+type Ticker struct {
+	C    <-chan time.Time
+	done chan struct{}
+}
+
+// NewTicker returns a new Ticker containing a channel that will send
+// the time with a period specified by the duration argument, similar to
+// standard library's time.Ticker
+//
+// It differs from the standard library Ticker by closing C when
+// Stop() is called, which allows for ranging over C:
+//
+//		for range ticker.C {
+//   		// do something
+//		}
+func NewTicker(d time.Duration) *Ticker {
+	c := make(chan time.Time)
+	done := make(chan struct{})
+	tt := time.NewTicker(d)
+	go func() {
+		for {
+			select {
+			case tick := <-tt.C:
+				c <- tick
+			case <-done:
+				tt.Stop()
+				close(c)
+				return
+			}
+		}
+	}()
+	return &Ticker{C: c, done: done}
+}
+
+// Stop turns off a ticker. After Stop, no more ticks will be sent. Stop
+// closes the channel. It save to call Stop more than once.
```

Comment:
4 short sentences is a bit stilted. Perhaps word this as:

> Stop turns off the ticker, closing the ticker channel C, after which no more ticks will be sent. It is safe to call Stop more than once; subsequent calls to Stop do nothing.

Reply:
done.